eva_sub_cli_processing/process_jobs.py

```diff
@@ -53,47 +30,105 @@ def _process_submission(submission):
 class SubmissionScanner(AppLogger):
 
     statuses = []
+    step_statuses = []
 
-    def scan(self):
+    def _scan_per_status(self):
         submissions = []
         for status in self.statuses:
-            for submission_data in _get_submission_api(_url_build('admin', 'submissions', 'status', status)):
-                submissions.append(Submission(
+            for submission_data in get_from_sub_ws(sub_ws_url_build('admin', 'submissions', 'status', status)):
+                submissions.append(SubmissionStep(
                     submission_id=submission_data.get('submissionId'),
-                    submission_status=submission_data.get('status'),
-                    uploaded_time=submission_data.get('uploadedTime')
+                    status=submission_data.get('status'),
+                    processing_step=VALIDATION,
+                    processing_status=READY_FOR_PROCESSING,
+                    last_update_time=submission_data.get('uploadedTime'),
+                    priority=5
                 ))
         return submissions
 
+    def _scan_per_step_status(self):
+        submissions = []
+        for step, status in self.step_statuses:
+            for submission_step_data in get_from_sub_ws(sub_ws_url_build('admin', 'submission-processes', step, status)):
+                submissions.append(SubmissionStep(
+                    submission_id=submission_step_data.get('submissionId'),
+                    status=PROCESSING,
+                    processing_step=step,
+                    processing_status=status,
+                    last_update_time=submission_step_data.get('lastUpdateTime'),
+                    priority=submission_step_data.get('priority')
+                ))
+        return submissions
+
+    def scan(self):
+        return self._scan_per_status() + self._scan_per_step_status()
+
     def report(self):
-        header = ['Submission Id', 'Submission status', 'Uploaded time']
+        header = ['Submission Id', 'Submission status', 'Processing step', 'Processing status', 'Last updated time',
+                  'Priority']
         scan = self.scan()
         lines = []
         for submission in scan:
-            lines.append((submission.submission_id, submission.submission_status, str(submission.uploaded_time)))
+            lines.append((submission.submission_id, submission.submission_status, submission.processing_step,
+                          submission.processing_status, str(submission.last_update_time), str(submission.priority)))
         pretty_print(header, lines)
 
 
 class NewSubmissionScanner(SubmissionScanner):
 
     statuses = ['UPLOADED']
+    step_statuses = []
```

Comment:
This is mostly a matter of taste, but I think I would prefer the different scanning tasks as methods rather than classes.  So we would have just one `SubmissionScanner` with the generic `_scan_per_step_status` helper method, and `find_new_submissions`, `find_completed_submission_steps`, etc. that call that method with the relevant statuses.

On the other hand, maybe there's more functionality that would go into these subclasses that I'm not thinking of, in which case having the extra classes makes sense.